qiskit_experiments/calibration_management/calibrations.py

```diff
@@ -93,6 +132,270 @@ def __init__(self, control_config: Dict[Tuple[int, ...], List[ControlChannel]] =
+        self._register_parameter(self.qubit_freq, ())
+        self._register_parameter(self.meas_freq, ())
+
+        # Backends like Armonk do not have a coupling map.
+        num_qubits = max(max(coupling_map)) + 1 if coupling_map is not None else 1
+
+        self._qubits = list(range(num_qubits))
+        self._coupling_map = coupling_map
+
+        self.update_inst_map_on_add = True
+
+        # Push the schedules to the instruction schedule map.
+        self.update_inst_map()
+
+    @classmethod
+    def from_backend(
+        cls,
+        backend: Backend,
+        library: Optional[BasisGateLibrary] = None,
+        add_parameter_defaults: bool = True,
+    ) -> "Calibrations":
+        """Create an instance of Calibrations from a backend.
+
+        Args:
+            backend: A backend instance from which to extract the qubit and readout frequencies
+                (which will be added as first guesses for the corresponding parameters) as well
+                as the coupling map. If the backend is not given then the calibrations will be
+                initialized with the following key-word arguments: control_config, coupling_map,
+                num_qubits.
+            library: A library class that will be instantiated with the library options to then
+                get template schedules to register as well as default parameter values.
+            add_parameter_defaults: A boolean to indicate whether the default parameter values of
+                the given library should be used to populate the calibrations. By default this
+                value is ``True`` but can be set to ``False`` when deserializing a calibrations
+                object.
+
+        Returns:
+            An instance of Calibrations instantiated from a backend.
+        """
+        cals = Calibrations(
+            getattr(backend.configuration(), "coupling_map", None),
+            getattr(backend.configuration(), "control_channels", None),
+            library,
+            add_parameter_defaults,
+        )
+
+        # Disable the update since we will be adding several parameters.
+        cals.update_inst_map_on_add = False
+
+        if add_parameter_defaults:
+            for qubit, freq in enumerate(getattr(backend.defaults(), "qubit_freq_est", [])):
+                cals.add_parameter_value(freq, cals.qubit_freq, qubit)
+
+            for meas, freq in enumerate(getattr(backend.defaults(), "meas_freq_est", [])):
+                cals.add_parameter_value(freq, cals.meas_freq, meas)
+
+        cals.update_inst_map_on_add = True
+
+        # Now update the instruction schedule map.
+        cals.update_inst_map()
+
+        return cals
+
+    @property
+    def library(self) -> Optional[BasisGateLibrary]:
+        """Return the name of the library, e.g. for experiment metadata."""
+        return self._library
+
+    @property
+    def default_inst_map(self) -> InstructionScheduleMap:
```

Comment:
default refers to group? Since `get_inst_map` defaults to `default` I don't have strong motivation to have dedicated property. i.e. we can add branching logic to return `self._inst_map` if group is default and no cutoff date.

Reply:
Done here: 84bdc1f